### PR TITLE
Split out SSL init from https_get

### PR DESCRIPTION
**Pull request by acv**
This is the fix for #119

### CODE REVIEW

**Comment by acv** [comment]
There's no clean up code for CYASSL_CTX. But it should only ever be initialized once and is needed right until the gateway exits and free all it's resources anyway.

**Comment by acv** [comment]
Before merging and once OK'd, I'll run the indent command line and merge that in as well.

File: src/simple_http.c [hunk 1]
```diff
@@ -119,8 +125,68 @@ int http_get(const int sockfd, char *buf) {
...
+    CYASSL_CTX *ret;
+	s_config *config = config_get_config();
+
+    LOCK_CYASSL_CTX();
+    
+    if (NULL == cyassl_ctx) {
+        CyaSSL_Init();
+        /* Create the CYASSL_CTX */
+        /* Allow SSLv3 up to TLSv1.2 */
+        if ( (cyassl_ctx = CyaSSL_CTX_new(CyaSSLv23_client_method())) == NULL){
```
Comment (mhaas) [inline src/simple_http.c]: Hum, this should only enable TLS 1.1 and up. SSL3 is broken. ISTR that I removed this, but I probably lost the commit. Not your fault here, but might as well change it.

**Comment by mhaas** [comment]
Looks good to me. I wonder if static analysers will (false-positively) complain about the non-freed resources.

**Comment by acv** [comment]
It shouldn't because it's held by a global and unless they solved the halting problem, they cannot tell when it will not be live anymore.

Alex

**Comment by acv** [comment]
Really need to re-indent conf.c/conf.h too now...

**Comment by mhaas** [comment]
Just re-indent the whole source tree?

**Comment by acv** [comment]
Let's create an issue for that then. And let's try to not have big merges before...

**Comment by mhaas** [comment]
Oh, good job on the configurable cipher suite!

**Pull request merged**